rust/core-lib/src/tabs.rs

```diff
@@ -371,9 +372,13 @@ impl<W: Write + Send + 'static> Documents<W> {
             let buffer_id = self.next_buffer_id();
             self.new_empty_view(rpc_peer, &view_id, buffer_id);
         }
+
+        // schedule plugin notification for after the view has received an identifer
+        let view_id2 = view_id.to_owned();
         let init_info = self.buffers.lock().editor_for_view(&view_id)
             .unwrap().plugin_init_info();
-        self.plugins.document_new(&view_id, init_info);
+        self.plugins.run_on_idle(move |p| p.document_new(&view_id2, &init_info));
```

Comment:
That indeed would be the simplest solution, but I don't like how it adds unnecessary complexity to the frontend API to fix a core plumbing problem.

Also re your atomics locking on every iteration point: I just thought about it and a lock is basically implemented as an operation on an atomic, which in the case of no contention (which will never happen with our single-threaded event loop, I don't even quite understand why a Mutex is needed) is very very cheap. So I wouldn't expect the atomic would improve performance much.

So at the very least I think it should be a `run_next_iteration` method on some kind of handle to the event loop (maybe using an unbounded `mpsc` queue, which gives you your queue in the same method as synchronization).

Even if it doesn't crop up that often, or only once, I still think the correct approach though is to make the event loop behave [what I think is] correctly with respect to notifications and RPCs with returns. At a stretch I can even imagine other resource creation cases like creating breakpoints. It might even be simpler than the `run_next_iteration` code, depending on how notifications actually work (I haven't looked).